gaia-header.js

```diff
@@ -305,27 +669,43 @@ module.exports = Component.register('gaia-header', {
   }
 
   /**
-   * .flush-left
+   * .cant-center
    *
-   * When the fitted text is flush with the
-   * edge of the left edge of the container
-   * we pad it in a bit.
+   * In the rare event that there isn't
+   * enough room to center, it's best we
+   * fallback to left alignment, otherwise
+   * the title just looks broken.
    */
 
-  ::content h1.flush-left {
-    padding-left: 10px;
+  ::content h1.cant-center {
+    text-align: start;
+    padding: 0;
+  }
+
+  /**
+   * .cant-center:first-child
+   *
+   * When the title is the first visible
+   * child (ie. no after any buttons) and
+   * it can't be centered we add some
+   * padding-start for aesthetics.
+   */
+
+  :host(:not([action])) h1.cant-center {
+    -moz-padding-start: 10px;
   }
 
   /**
-   * .flush-right
+   * .cant-center
    *
-   * When the fitted text is flush with the
-   * edge of the right edge of the container
-   * we pad it in a bit.
+   * When the title is the last visible child
+   * (ie. it's not before any buttons) and
+   * it can't be centered we add some
+   * padding-end for aesthetics.
    */
 
-  ::content h1.flush-right {
-    padding-right: 10px; /* 1 */
+  ::content h1.cant-center:nth-last-child(2) {
```

Comment:
> What dynamically add a class to state that the h1 is the 'last visible child'?

since we use scoped CSS here, let's not spend time on this comment :)

That said I wonder why we add this padding for this case only ? Because from what I understand, it would not be very nice either if the title stays too close to a button?